lib/auth/grpcserver.go

```diff
@@ -792,6 +792,23 @@ func (g *GRPCServer) CreateAccessRequest(ctx context.Context, req *types.AccessR
 	if err := services.ValidateAccessRequest(req); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	localClusterName, err := auth.GetClusterName()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	for _, resourceID := range req.Spec.RequestedResourceIDs {
+		if resourceID.ClusterName != "" {
+			if resourceID.ClusterName == localClusterName.GetClusterName() {
+				continue
+			}
+			if _, err := auth.GetRemoteCluster(resourceID.ClusterName); err == nil {
+				continue
+			}
+			return nil, trace.NotFound("remote cluster %q is not found", resourceID.ClusterName)
```

Comment:
This error message is confusing. For example, why does it mention "remote cluster"? A couple suggestions:
- Make error message more user-friendly, for example "access request contains invalid cluster name xxx".
- Instead of returning on the first error, can we validate all cluster names provided in the request and include them in the message?